mxcubeweb/core/components/lims.py

```diff
@@ -326,7 +322,7 @@ def select_proposal(self, proposal):
             )
 
             todays_session = HWR.beamline.lims.get_todays_session(
-                proposal_info, create_session=False
+                proposal_info, create_session=True
```

Comment:
Hm, the sessions should already exist when its being selected, at least with the current logic. `select_proposal` will for instance be called when the application is fetched so creating a session here would break that logic. A session otherwise created above: 

https://github.com/mxcube/mxcubeweb/pull/1385/files#diff-3c54a0ef3e4a4efd2c578368ac1933aa853c8f8553112067cc4c75058a25b476R242

The problem is maybe in your case that you also call this logic when a user selects a proposal when logging in as user ? In that case, would it be possible to add an argument `create_proposal` from that route. I imagine its post to `/porposal` ?

Reply:
`get_todays_session` only creates a session if it does not exist. So in my view, having a True here is safe in any situation (at this point the user has made a decision, the proposal to be used).

the https://github.com/mxcube/mxcubeweb/blob/develop/mxcubeweb/core/components/user/usermanager.py#L331 has the flag False always, so in our case the session is not going to be created, plus it would have been created for potentially a proposal that is not the one to be selected just after

sure we can add an argument to the call, but I still dont get why... 

Are you using user account based login? Maybe easier if we discuss this during tomorrow's meeting

Reply:
We are starting to use the user based logins and we are currently evaluating the SSO integration. There will soon be PR with our draft implementation, and we will see exactly how to continue from there. 

Regarding the above, Its not entirely clear to me and it might need some testing to verify. I think one issue is that the call to get_proposal_info, above does not check with LIMS for created proposals after login, https://github.com/mxcube/mxcubeweb/blob/a8e938ab56958d3a5f29e1c3b0c3e0425b5538d9/mxcubeweb/core/components/lims.py#L304

So that `proposal_info` is not updated and a new session is created.

Meaning that, I guess, any call to `get_todays_session`, would create a new session. The session I guess would be created on the LIMS side but would not be seen by mxcube until a new login is made. I think that is why the flag is `false`.

The logic here is a bit fragile and we've not reworked this as we are also working on the SSO integration.

I believe you if you say that youv'e tested and that it works